### PR TITLE
Feature for situation management

### PR DESCRIPTION
**Pull request by nyuuyn**
#### Short Description
Enables to use the situation api of a opentosca runtime within the ui

#### Proposed Changes

  * adds a situation tab that enables the following:
  * - creating situations
  * - creating (event) triggers on the situations by combining situations with plans
  * - manual triggering of situations

### CODE REVIEW

**Comment by nyuuyn** [comment]
How to use the view to configure different triggers which can either be started by users or external system via emitting an event:

Add a new situation:
Bildschirmfoto 2020-12-01 um 13 04 52
Bildschirmfoto 2020-12-01 um 13 04 56

Add a new trigger:
Bildschirmfoto 2020-12-01 um 13 05 19
Bildschirmfoto 2020-12-01 um 13 05 25

Trigger the change of the previously added situation:
Bildschirmfoto 2020-12-01 um 13 05 28

File: src/app/app.module.ts [hunk 1]
```diff
@@ -16,6 +16,8 @@ import { NgModule } from '@angular/core';
 import { FormsModule } from '@angular/forms';
 import { AppComponent } from './app.component';
 import { AboutComponent } from './about.component';
+//
```
Comment (zimmerml) [inline src/app/app.module.ts:19]: remove

File: src/app/repository/repository.component.ts [hunk 2]
```diff
@@ -63,6 +63,7 @@ export class RepositoryComponent implements OnInit {
     ngOnInit() {
         this.repositoryItems$.subscribe(items => this.repositoryItems = items);
         this.selectedRepository$.subscribe(item => {
+            //
```
Comment (zimmerml) [inline src/app/repository/repository.component.ts:66]: remove

**Comment by zimmerml** [comment]
image
The spacing between the buttons do not seem to fit

**Review by zimmerml — APPROVED** [review]
lgtm

File: src/app/repository/repository.component.ts (hunk 2 above)
Comment (nyuuyn) [reply src/app/repository/repository.component.ts]: fixed

File: src/app/app.module.ts (hunk 1 above)
Comment (nyuuyn) [reply src/app/app.module.ts]: fixed

**Pull request merged**